app/src/main/java/com/sdp/movemeet/models/FirebaseObject.java

```diff
@@ -3,16 +3,24 @@
 /**
  * An object that can be found in the Firebase Realtime Database/Firestore backend
  */
-public interface FirebaseObject {
```

Comment:
What's the advantage of using an abstract class instead of an interface?

Reply:
WIth an abstract class, we get to have a default implementation for the `getDocumentPath()` and `setDocumentPath()` methods, and as I was implementing FirebaseObjects I realized they were often identical codeblocks across classes. Hence, the change to an abstract class with a single implementation

Reply:
That's in fact a good point from you Kepler to have changed `FirebaseObject` from interface to an abstract class because you can in fact now create functionality that subclasses (such as User, Activity and Message) can implement or override! On the contrary, an interface only allows you to define functionalities, not implement them.